yjit/src/cruby.rs

```diff
@@ -301,6 +303,92 @@ pub struct rb_cref_t {
     _marker: core::marker::PhantomData<(*mut u8, core::marker::PhantomPinned)>,
 }
 
+/// Represents flags set for a callsite
+/// Comes from the rb_callinfo struct
+pub struct CallFlags(c_uint);
+
+impl CallFlags {
+    pub fn from_ci(ci: *const rb_callinfo) -> Self {
+        Self(unsafe { vm_ci_flag(ci) })
+    }
+
+    pub fn is_kw_splat(&self) -> bool {
+        self.0 & VM_CALL_KW_SPLAT != 0
+    }
+
+    pub fn is_blockarg(&self) -> bool {
+        self.0 & VM_CALL_ARGS_BLOCKARG != 0
+    }
+
+    pub fn is_kwarg(&self) -> bool {
+        self.0 & VM_CALL_KWARG != 0
+    }
+
+    pub fn is_splat(&self) -> bool {
```

Comment:
```suggestion
    pub fn is_args_blockarg(&self) -> bool {
        self.0 & VM_CALL_ARGS_BLOCKARG != 0
    }

    pub fn is_kwarg(&self) -> bool {
        self.0 & VM_CALL_KWARG != 0
    }

    pub fn is_args_splat(&self) -> bool {
```

As it's a JIT implementation, we will always need to compare YJIT's code against the VM's code. I think it's useful for maintainability to keep consistency with the baseline implementation as much as possible rather than only focusing on code shortness. 

If we want, since it's not exposed as C API, we could always propose to remove `ARGS_` from the original constants as well. Either way, I want code for the JIT implementation to be consistent with the target interpreter's one.